### DISCUSSION TITLE
Starts then stops after transmission is up

### DISCUSSION THREAD

**Discussion by LeBrunton** [haugene/docker-transmission-openvpn#1965] [Q&A] [answered]
Hi,

I am trying to run this docker on a QNAP NAS and I almost have it working, however just after Transmission comes up I can even access it briefly, the container stops.

Here is my log file:

Starting container with revision: 6ee67f8210d7f9ca17677c2b108390acc91ec5ab

Creating TUN device /dev/net/tun

Using OpenVPN provider: NORDVPN

Running with VPN_CONFIG_SOURCE auto

Provider NORDVPN has a bundled setup script. Defaulting to internal config

Executing setup script for NORDVPN

2021-09-24 16:10:20 Checking curl installation

2021-09-24 16:10:20 Removing existing configs

2021-09-24 16:10:20 Selecting the best server...

2021-09-24 16:10:20 Searching for group: legacy_p2p

2021-09-24 16:10:20 Searching for technology: openvpn_udp

2021-09-24 16:10:20 Best server : uk1823.nordvpn.com

2021-09-24 16:10:20 Downloading config: default.ovpn

2021-09-24 16:10:20 Downloading from: https://downloads.nordcdn.com/configs/files/ovpn_udp/servers/uk1823.nordvpn.com.udp.ovpn

2021-09-24 16:10:20 Selecting the best server...

2021-09-24 16:10:20 Searching for group: legacy_p2p

2021-09-24 16:10:20 Searching for technology: openvpn_udp

2021-09-24 16:10:20 Best server : uk2229.nordvpn.com

2021-09-24 16:10:20 Downloading config: uk2229.nordvpn.com.ovpn

2021-09-24 16:10:20 Downloading from: https://downloads.nordcdn.com/configs/files/ovpn_udp/servers/uk2229.nordvpn.com.udp.ovpn

Starting OpenVPN using config uk2229.nordvpn.com.ovpn

Modifying /etc/openvpn/nordvpn/uk2229.nordvpn.com.ovpn for best behaviour in this container

Modification: Point auth-user-pass option to the username/password file

Modification: Change ca certificate path

Modification: Change ping options

Setting OpenVPN credentials...

adding route to local network 192.168.1.0/24 via 172.29.12.1 dev eth0

Fri Sep 24 16:10:21 2021 OpenVPN 2.4.7 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] built on Jul 19 2021

Fri Sep 24 16:10:21 2021 library versions: OpenSSL 1.1.1f  31 Mar 2020, LZO 2.10

Fri Sep 24 16:10:21 2021 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts

Fri Sep 24 16:10:21 2021 Outgoing Control Channel Authentication: Using 512 bit message hash 'SHA512' for HMAC authentication

Fri Sep 24 16:10:21 2021 Incoming Control Channel Authentication: Using 512 bit message hash 'SHA512' for HMAC authentication

Fri Sep 24 16:10:21 2021 TCP/UDP: Preserving recently used remote address: [AF_INET]178.239.162.207:1194

Fri Sep 24 16:10:21 2021 Socket Buffers: R=[1048576->1048576] S=[1048576->1048576]

Fri Sep 24 16:10:21 2021 UDP link local: (not bound)

Fri Sep 24 16:10:21 2021 UDP link remote: [AF_INET]178.239.162.207:1194

Fri Sep 24 16:10:21 2021 TLS: Initial packet from [AF_INET]178.239.162.207:1194, sid=a88029b2 7abc72b8

Fri Sep 24 16:10:21 2021 WARNING: this configuration may cache passwords in memory -- use the auth-nocache option to prevent this

Fri Sep 24 16:10:21 2021 VERIFY OK: depth=2, C=PA, O=NordVPN, CN=NordVPN Root CA

Fri Sep 24 16:10:21 2021 VERIFY OK: depth=1, C=PA, O=NordVPN, CN=NordVPN CA6

Fri Sep 24 16:10:21 2021 VERIFY KU OK

Fri Sep 24 16:10:21 2021 Validating certificate extended key usage

Fri Sep 24 16:10:21 2021 ++ Certificate has EKU (str) TLS Web Server Authentication, expects TLS Web Server Authentication

Fri Sep 24 16:10:21 2021 VERIFY EKU OK

Fri Sep 24 16:10:21 2021 VERIFY OK: depth=0, CN=uk2229.nordvpn.com

Fri Sep 24 16:10:21 2021 Control Channel: TLSv1.3, cipher TLSv1.3 TLS_AES_256_GCM_SHA384, 4096 bit RSA

Fri Sep 24 16:10:21 2021 [uk2229.nordvpn.com] Peer Connection Initiated with [AF_INET]178.239.162.207:1194

Fri Sep 24 16:10:23 2021 SENT CONTROL [uk2229.nordvpn.com]: 'PUSH_REQUEST' (status=1)

Fri Sep 24 16:10:23 2021 PUSH: Received control message: 'PUSH_REPLY,redirect-gateway def1,dhcp-option DNS 103.86.96.100,dhcp-option DNS 103.86.99.100,sndbuf 524288,rcvbuf 524288,explicit-exit-notify,comp-lzo no,route-gateway 10.8.3.1,topology subnet,ping 60,ping-restart 180,ifconfig 10.8.3.6 255.255.255.0,peer-id 4,cipher AES-256-GCM'

Fri Sep 24 16:10:23 2021 OPTIONS IMPORT: timers and/or timeouts modified

Fri Sep 24 16:10:23 2021 OPTIONS IMPORT: explicit notify parm(s) modified

Fri Sep 24 16:10:23 2021 OPTIONS IMPORT: compression parms modified

Fri Sep 24 16:10:23 2021 OPTIONS IMPORT: --sndbuf/--rcvbuf options modified

Fri Sep 24 16:10:23 2021 Socket Buffers: R=[1048576->1048576] S=[1048576->1048576]

Fri Sep 24 16:10:23 2021 OPTIONS IMPORT: --ifconfig/up options modified

Fri Sep 24 16:10:23 2021 OPTIONS IMPORT: route options modified

Fri Sep 24 16:10:23 2021 OPTIONS IMPORT: route-related options modified

Fri Sep 24 16:10:23 2021 OPTIONS IMPORT: --ip-win32 and/or --dhcp-option options modified

Fri Sep 24 16:10:23 2021 OPTIONS IMPORT: peer-id set

Fri Sep 24 16:10:23 2021 OPTIONS IMPORT: adjusting link_mtu to 1657

Fri Sep 24 16:10:23 2021 OPTIONS IMPORT: data channel crypto options modified

Fri Sep 24 16:10:23 2021 Data Channel: using negotiated cipher 'AES-256-GCM'

Fri Sep 24 16:10:23 2021 Outgoing Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key

Fri Sep 24 16:10:23 2021 Incoming Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key

Fri Sep 24 16:10:23 2021 ROUTE_GATEWAY 172.29.12.1/255.255.252.0 IFACE=eth0 HWADDR=02:42:ac:1d:0c:02

Fri Sep 24 16:10:23 2021 TUN/TAP device tun0 opened

Fri Sep 24 16:10:23 2021 TUN/TAP TX queue length set to 100

Fri Sep 24 16:10:23 2021 /sbin/ip link set dev tun0 up mtu 1500

Fri Sep 24 16:10:23 2021 /sbin/ip addr add dev tun0 10.8.3.6/24 broadcast 10.8.3.255

Fri Sep 24 16:10:23 2021 /etc/openvpn/tunnelUp.sh tun0 1500 1585 10.8.3.6 255.255.255.0 init

Up script executed with tun0 1500 1585 10.8.3.6 255.255.255.0 init

Updating TRANSMISSION_BIND_ADDRESS_IPV4 to the ip of tun0 : 10.8.3.6

Updating Transmission settings.json with values from env variables

Generating settings.json for Transmission from environment and defaults /etc/transmission/default-settings.json

Overriding bind-address-ipv4 because TRANSMISSION_BIND_ADDRESS_IPV4 is set to 10.8.3.6

Overriding download-dir because TRANSMISSION_DOWNLOAD_DIR is set to /data/completed

Overriding incomplete-dir because TRANSMISSION_INCOMPLETE_DIR is set to /data/incomplete

Overriding rpc-port because TRANSMISSION_RPC_PORT is set to 9091

Overriding watch-dir because TRANSMISSION_WATCH_DIR is set to /data/watch

sed'ing True to true

-------------------------------------

Transmission will run as

-------------------------------------

User name:   root

User uid:    0

User gid:    0

-------------------------------------

STARTING TRANSMISSION

Transmission startup script complete.

Fri Sep 24 16:10:23 2021 /sbin/ip route add 178.239.162.207/32 via 172.29.12.1

Fri Sep 24 16:10:23 2021 /sbin/ip route add 0.0.0.0/1 via 10.8.3.1

Fri Sep 24 16:10:23 2021 /sbin/ip route add 128.0.0.0/1 via 10.8.3.1

Fri Sep 24 16:10:23 2021 Initialization Sequence Completed

I can't see that there is a final error message to indicate what went wrong, any suggestions?

Thanks
Lewis

**Comment by pkishino** [comment]
I do recall a previous issue relating to this that mentioned that qnap had some problem with docker and openvpn. Some watchdog kills the process or some such. I’d start there

**Reply by pkishino** [reply]
Couldn’t see any problems in the log so..

**Comment by LeBrunton** [comment, marked as answer]
Ah that’s not good. All the posts I can find across the net at the moment suggest that there is no fix, at least not with the latest version of the weak firmware :(
